### PR TITLE
Move the experimental frontends into a separate crate, so that when not using them they don't take several minutes to compile (and indirect dependencies).

### PR DESCRIPTION
**Pull request by arnaucube**
Move the experimental frontends into a separate crate, so that when not using them they don't take several minutes to compile (and indirect dependencies).

This saves several minutes (and MBs of data) on compilation time both when running tests in this repo, but also when using the sonobe lib as a dependency in external repos.

This solution was suggested by @ed255 when talking about this.

### CODE REVIEW

**Review by pmikolajczyk41 — COMMENTED** [review]
Please accept apologies for interrupting, but have you considered hiding each frontend behind a dedicated feature (and making deps optional)?

Edit: I see you have :) can you share the reasons to abandon features in favour of crates?

**Comment by arnaucube** [comment]
Indeed, that's a good point. This was initially done at https://github.com/privacy-scaling-explorations/sonobe/pull/141 but the issue is that even if a frontend was 'not used' by not using its feature, it's dependencies were still being imported (and the dependencies of the dependencies, taking long minutes and MBs of data), when running tests but also when using Sonobe as external dependency.

With the separation into a different subpackage for the experimental frontends done in this PR, the users of the lib that don't use any of the experimental frontends will not get impacted by those dependencies that are in fact not used in that case, and only in the cases where the experimental frontends are specifically used those dependencies will get used.

The recommended frontend is using arkworks (directly implementing the [FCircuit trait](https://github.com/privacy-scaling-explorations/sonobe/blob/main/folding-schemes/src/frontend/mod.rs)), and the other experimental frontends add overhead when generating the witness which makes the folding step slower.

**Comment by pmikolajczyk41** [comment]
oh, okay; now I remember similar issue I had with private repo dependency in a workspace causing problems for every member crate, even for those, that didn't use it (even indirectly); afair a workaround it was to use some dummy patching, but in Sonobe case I totally understand and support your final decision - thanks!

**Review by CPerezz — APPROVED** [review]
LGTM

File: folding-schemes/src/frontend/utils.rs
Comment (arnaucube) [reply folding-schemes/src/frontend/utils.rs]: my bad, I disabled it to run some local tests using the lib as a dependency to ensure that everything works fine, so needed that circuit exposed to external usage. But just [re-enabled](https://github.com/privacy-scaling-explorations/sonobe/compare/27ab0e2a41f62d7c658f2a0450332593878d429b..2bb66c1fcdfd19e9bfa7b516f5becfa9782104ae) it again.

File: frontends/Cargo.toml
Comment (arnaucube) [reply frontends/Cargo.toml:25]: which feature from the folding-schemes crate would you like to select from the frontend crate?

File: folding-schemes/src/frontend/utils.rs
Comment (arnaucube) [reply folding-schemes/src/frontend/utils.rs]: my bad, this is because in the crate frontends, the WrapperCircuit is used for some tests, so with exposing it we avoid duplicating it. Updated it to [completely remove the lines](https://github.com/privacy-scaling-explorations/sonobe/compare/2bb66c1fcdfd19e9bfa7b516f5becfa9782104ae..b42999f507c823e97fa1361be3b04986d6804a36)

File: frontends/README.md
Comment (dmpierre) [inline frontends/README.md:11]: We should warn that noname is partially supported, as suggested in #150 . We can then also close this issue.

**Review by dmpierre — APPROVED** [review]
One suggestion regarding the `README.md` of the frontend crate. Otherwise lgtm!

File: README.md
Comment (CPerezz) [reply README.md:36]: I know I know. But seems good to just have some small mention in the README. Which can help finding about the frontends readme which is more specialized.

File: frontends/Cargo.toml
Comment (CPerezz) [reply frontends/Cargo.toml:25]: `parallel` and unsure about `light_test`. 

Although we can just ship parallel by default always. I just don't know if it makes sense.. I assume so for now. WASM doesn't have any issues with it. So fine for me.

Comment (arnaucube) [reply frontends/Cargo.toml:25]: but the parallel affects the frontends crate? The `frontends` crate only imports the `folding-schemes` crate to use its `FCircuit` trait (and the PathOrBin & Error), which does not involve `parallel` nor the `light-test` stuff.

Comment (CPerezz) [reply frontends/Cargo.toml:25]: Ohh fair point. 

Resolving this.

Comment (arnaucube) [reply frontends/Cargo.toml:25]: ok, will leave it as it this.

**Comment by arnaucube** [comment]
Will wait till https://github.com/privacy-scaling-explorations/sonobe/pull/169 is merged, and then will rebase this PR (#168) to main (once it includes #169) and deal with the git conflicts in this branch (#168).

**Comment by CPerezz** [comment]
Let's just merge this @arnaucube as I want to wait for Janabel to confirm that the PR works prev to merge it.

File: frontends/README.md
Comment (arnaucube) [reply frontends/README.md]: Cool, added the mention ^^

File: README.md
Comment (arnaucube) [reply README.md:36]: cool, added

**Pull request merged**
